implementations/rust/ockam/ockam_core/src/api.rs

```diff
@@ -523,43 +531,50 @@ impl<T: Encode<()>> ResponseBuilder<T> {
 }
 
 #[allow(unused_variables)]
-pub fn assert_request_match<'a>(struct_name: impl Into<Option<&'a str>>, cbor: &[u8]) {
-    #[cfg(feature = "tag")]
```

Comment:
> Regarding the cfg tag flag. That is just used for cddl related stuff right? I removed if for now, because it was causing some issues, but I can add it again later. For example the assert_request_match doesn't do anything when the flag is disabled right?

We have a [`tag` field](https://github.com/build-trust/ockam/blob/495c3a04bc20081fc21220f054a1f38d5c199e19/implementations/rust/ockam/ockam_api/src/cloud/addon.rs#L15) on all cbor structs that can be used to identify the struct and it's enabled with that flag.

I assume we have to somehow avoid validating that field when the `tag` feature is disabled.

The best thing would be to have some unit tests to check that the validation works as expected with and without the `flag` feature.